ui/ducks/metamask/metamask.test.js

```diff
@@ -426,4 +462,79 @@ describe('MetaMask Reducers', () => {
       ).toBe(false);
     });
   });
+
+  describe('getGasFeeControllerEstimates', () => {
+    it('returns estimates from GasFeeController state', () => {
+      const state = {
+        metamask: {
+          gasFeeEstimates: GAS_FEE_CONTROLLER_ESTIMATES_MOCK,
+        },
+      };
+
+      expect(getGasFeeControllerEstimates(state)).toStrictEqual(
+        GAS_FEE_CONTROLLER_ESTIMATES_MOCK,
+      );
+    });
+  });
+
+  describe('getTransactionGasFeeEstimates', () => {
+    it('returns estimates from transaction', () => {
+      const state = {
+        confirmTransaction: {
+          txData: {
+            gasFeeEstimates: TRANSACTION_ESTIMATES_MOCK,
+          },
+        },
+      };
+
+      expect(getTransactionGasFeeEstimates(state)).toStrictEqual(
+        TRANSACTION_ESTIMATES_MOCK,
+      );
+    });
+
+    it('returns undefined if no confirm transaction', () => {
+      expect(getTransactionGasFeeEstimates({})).toBeUndefined();
+    });
+  });
+
+  describe('getGasFeeEstimates', () => {
+    it('returns GasFeeController estimates if no transaction estimates', () => {
+      const state = {
+        metamask: {
+          gasFeeEstimates: GAS_FEE_CONTROLLER_ESTIMATES_MOCK,
+        },
+      };
+
+      expect(getGasFeeEstimates(state)).toStrictEqual(
+        GAS_FEE_CONTROLLER_ESTIMATES_MOCK,
+      );
+    });
+
+    it('returns merged transaction estimates if transaction estimates and GasFeeController estimates', () => {
+      const state = {
+        confirmTransaction: {
+          txData: {
+            gasFeeEstimates: TRANSACTION_ESTIMATES_MOCK,
+          },
+        },
+        metamask: {
+          gasEstimateType: GAS_ESTIMATE_TYPES.FEE_MARKET,
+          gasFeeEstimates: GAS_FEE_CONTROLLER_ESTIMATES_MOCK,
+        },
+      };
+
+      mergeGasFeeEstimatesMock.mockReturnValue(TRANSACTION_ESTIMATES_MOCK);
+
+      expect(getGasFeeEstimates(state)).toStrictEqual(
+        TRANSACTION_ESTIMATES_MOCK,
```

Comment:
Using `TRANSACTION_ESTIMATES_MOCK` again might lead to confusion since it's part of the state. Suggesting `SENTINEL_ESTIMATES = {}` as a clear marker to show `getGasFeeEstimates` outputs what `mergeGasFeeEstimates` does, and use `expect(...).toBe(SENTINEL_ESTIMATES)` for clarity instead of `toStrictlyEqual`.

Reply:
My reasoning here was I didn't want to test the merge function again since it already had unit tests, but I though it would be more confusing if I didn't also setup the state for this scenario where the transaction estimates are in the state.